tests/test_codelist.py

```diff
@@ -1,203 +1,231 @@
+        )
+
+
+def test_variable_codelist_multiple_units():
+    """Check that multiple units work in a VariableCodeList"""
+    codelist = VariableCodeList.from_directory(
+        "variable", TEST_DATA_DIR / "multiple_unit_codelist"
+    )
+    assert codelist["Var1"].unit == ["unit1", "unit2"]
+
+
+def test_to_excel_read_excel_roundtrip(tmpdir):
+
+    codelist_dir = TEST_DATA_DIR / "variable_codelist_complex_attr"
+
+    # read VariableCodeList
+    exp = VariableCodeList.from_directory("variable", codelist_dir)
+    # save to temporary file
+    exp.to_excel(tmpdir / "output.xlsx")
+    # read from temporary file
+    obs = VariableCodeList.read_excel(
+        "variable",
+        tmpdir / "output.xlsx",
+        "variable",
+        "Variable",
+        attrs=["Description", "Unit", "Region-aggregation"],
+    )
+
+    assert obs.name == exp.name
+    # since the obs and exp are read from different files, the file attribute will be
+    # different. For comparison we remove it
+    assert remove_file_from_mapping(obs.mapping) == remove_file_from_mapping(
+        exp.mapping
+    )
+
+
+def test_to_yaml_from_directory(tmp_path):
+    """Test that creating a codelist from a yaml file and writing it to yaml produces
+    the same file"""
+
+    # read VariableCodeList
+    exp = VariableCodeList.from_directory(
+        "variable", TEST_DATA_DIR / "variable_codelist_complex_attr"
+    )
+    exp.to_yaml(tmp_path / "variables.yaml")
+
+    # read from temporary file
+    obs = VariableCodeList.from_directory("variable", tmp_path)
+
+    assert obs.name == exp.name
+    # since the obs and exp are read from different files, the file attribute will be
+    # different. For comparison we remove it
+    assert remove_file_from_mapping(obs.mapping) == remove_file_from_mapping(
+        exp.mapping
+    )
+    
+    
+def test_RegionCodeList_hierarchy_filter():
+    """Test that verifies the hierarchy filter can sort through list of regions and
+    give list of regions contained in the given hierarchy"""
+
+    # read RegionCodeList
+    rcl = RegionCodeList.from_directory("Region", TEST_DATA_DIR / "region_codelist")
+    filter_param = "countries"
+    obs = rcl.filter(filter_param)
+    extra_attributes = {'iso2': 'XY', 'iso3': 'XYZ', 'hierarchy': 'countries',
+                        'file': 'region_codelist/region.yaml'}
+    mapping = {'Some Country': Code(name='Some Country', description=None, 
+                                    extra_attributes=extra_attributes)}
+    exp = RegionCodeList(name=filter_param, mapping=mapping)
+    assert obs == exp
+    
+    
+def test_RegionCodeList_hierarchy_filter_ValueError():
+    """Test that verifies the filter gives error when user inputs an unrecognizeable
+    hierarchy"""
+    
+    # read RegionCodeList
+    rcl = RegionCodeList.from_directory("Region", TEST_DATA_DIR / "region_codelist")
+    match = "Hierarchy level does not exist: R77."

```

Comment:
I think this test is currently failing because `match` is a regular expression where "." means _match anything_.
If you leave out the dot it should be all good. If you actually want to match for the . character and not for anything you'd need to escape it. If I recall correctly that should work like this: "\." or "\\.".